lib/src/common/observer_controller.dart

```diff
@@ -501,34 +501,125 @@ mixin ObserverControllerForScroll on ObserverControllerForInfo {
-      // It is separator
-      final nextChildPaintBounds = nextChild.paintBounds;
-      final nextChildSize = isHorizontal
-          ? nextChildPaintBounds.width
-          : nextChildPaintBounds.height;
-      separatorTotalHeight = index * nextChildSize;
-    }
     final childPaintBounds = targetChild.paintBounds;
-    final childSize =
+    double itemSeparatorHeight = 0;
+    int indexLineCount = index;
+    if (obj is RenderSliverList || obj is RenderSliverFixedExtentList) {
+      // ListView
+      var nextChild = obj.childAfter(targetChild);
+      nextChild ??= obj.childBefore(targetChild);
+      if (nextChild != null && nextChild is! RenderIndexedSemantics) {
+        // It is separator
+        final nextChildPaintBounds = nextChild.paintBounds;
+        itemSeparatorHeight = isHorizontal
+            ? nextChildPaintBounds.width
+            : nextChildPaintBounds.height;
+      }
+    } else if (obj is RenderSliverGrid) {
+      // GirdView
+      double crossAxisSpacing = 0;
+      bool isHaveSetCrossAxisSpacing = false;
+      var nextChild = obj.childAfter(targetChild);
+      var nextChildOrigin =
+          nextChild?.localToGlobal(Offset.zero) ?? Offset.zero;
+      final targetChildOrigin = targetChild.localToGlobal(Offset.zero);
+      while (nextChild != null &&
+          (isHorizontal
+              ? nextChildOrigin.dx == targetChildOrigin.dx
+              : nextChildOrigin.dy == targetChildOrigin.dy)) {
+        // calculate crossAxisSpacing
+        if (!isHaveSetCrossAxisSpacing) {
+          if (isHorizontal) {
+            crossAxisSpacing =
+                (nextChildOrigin.dy - targetChildOrigin.dy).abs() -
+                    childPaintBounds.height;
+          } else {
+            crossAxisSpacing =
+                (nextChildOrigin.dx - targetChildOrigin.dx).abs() -
+                    childPaintBounds.width;
+          }
+          isHaveSetCrossAxisSpacing = true;
+        }
+        nextChild = obj.childAfter(nextChild);
+        nextChildOrigin = nextChild?.localToGlobal(Offset.zero) ?? Offset.zero;
+      }
+      if (nextChild != null) {
+        if (isHorizontal) {
+          itemSeparatorHeight =
+              (nextChildOrigin.dx - targetChildOrigin.dx).abs() -
+                  childPaintBounds.width;
+        } else {
+          itemSeparatorHeight =
+              (nextChildOrigin.dy - targetChildOrigin.dy).abs() -
+                  childPaintBounds.height;
+        }
+      } else {
+        var previousChild = obj.childBefore(targetChild);
+        var previousChildOrigin =
+            previousChild?.localToGlobal(Offset.zero) ?? Offset.zero;
+        while (previousChild != null &&
+            (isHorizontal
+                ? previousChildOrigin.dx == targetChildOrigin.dx
+                : previousChildOrigin.dy == targetChildOrigin.dy)) {
+          // calculate crossAxisSpacing
+          if (!isHaveSetCrossAxisSpacing) {
+            double crossAxisOriginX1 =
```

Comment:
I think the names of `crossAxisOriginX1` and `crossAxisOriginX2` should be changed. It does not only refer to the x-axis.

Reply:
you are right, i'll update the names